.github/workflows/check-learning-path-links.yml

```diff
@@ -0,0 +1,78 @@
+name: 'Check Learning Path Links'
+on:
+  schedule: # Run once a week
+    - cron: '0 0 * * 1'
+  workflow_dispatch:
+
+permissions: {}
+  
+jobs:
+  check-learning-path-links:
+    name: 'Check Learning Path Links'
+    runs-on: ubuntu-latest
+    permissions:
+      issues: write
+
+    steps:
+      - name: Checkout head
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: true
+          ref: main
+          path: head
+
+      - name: Get current SHA
+        id: get_new_sha
+        working-directory: ./head
+        run: |
+          new_sha=$(git rev-parse HEAD)
+          echo "new_sha=$new_sha" >> $GITHUB_ENV
+
+      - name: Get previous update SHA
+        id: get_sha
+        working-directory: ./head
+        run: |
+          cd .github
+          prev_sha=$(cat learning-path-sha.txt)
+          echo "prev_sha=$prev_sha" >> $GITHUB_ENV
+          cd ../
+    
+      - name: Checkout previous update
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+          ref:  ${{ env.prev_sha }}
+          path: prev
+
+      - name: Get changed files
+        working-directory: ./head
+        run: |
+          changed_source_files=$(git diff-tree --no-commit-id --name-only -r "$prev_sha" "$GITHUB_SHA" | { grep "**.cs$" || test $? = 1; })
+          echo "Files to validate: '${changed_source_files}'"
+          echo "updated_files=$(echo ${changed_source_files})" >> $GITHUB_ENV
+
+      - name: Check Learning Path Links
+        id: check-links
+        uses: ./.github/actions/learning-path-staleness-check
+        with:
+          repoURLToSearch: 'https://github.com/dotnet/dotnet-monitor'
+          learningPathsDirectory: 'documentation/learningPath'
+          changedFilePaths: ${{ env.updated_files }}
+          sourceDirectoryName: 'src'
+          oldHash: ${{ env.prev_sha }}
+          newHash: ${{ env.new_sha }}
+          
+      - name: Generate artifacts (Comment)
+        working-directory: ./head
+        run: |
+          mkdir -p ./learning-path-review
+          echo -n "${{ steps.check-links.outputs.modifiedFiles }}" > ./learning-path-review/modifiedFiles
```

Comment:
Using substitution expressions here are a script injection risk. All of these should be passed in via an `env` block instead to avoid the risk.

Example:
https://github.com/dotnet/dotnet-monitor/blob/341ff64a6097fe0bc66950e254e6160abcf77b84/.github/workflows/update-release-version.yml#L89-L96